packages/ra-ui-materialui/src/field/ReferenceManyField.js

```diff
@@ -106,22 +106,65 @@ ReferenceManyFieldView.propTypes = {
  *    ...
  * </ReferenceManyField>
  */
-export const ReferenceManyField = ({ children, ...props }) => {
+export const ReferenceManyField = ({
+    children,
+    sort: initialSort,
+    perPage: initialPerPage,
+    resource,
+    reference,
+    record,
+    target,
+    filter,
+    source,
+    basePath,
+    ...props
+}) => {
     if (React.Children.count(children) !== 1) {
         throw new Error(
             '<ReferenceManyField> only accepts a single child (like <Datagrid>)'
         );
     }
+    const { sort, setSort } = useSort(initialSort);
+    const { page, perPage, setPage, setPerPage } = usePagination(initialPerPage);
+
+    const {
+        data,
+        ids,
+        loadedOnce,
+        referenceBasePath,
+        total,
+    } = useReferenceMany({
+        resource,
+        reference,
+        record,
+        target,
+        filter,
+        source,
+        basePath,
+        page,
+        perPage,
+        sort,
+    });
 
     return (
-        <ReferenceManyFieldController {...props}>
-            {controllerProps => (
-                <ReferenceManyFieldView
-                    {...props}
-                    {...{ children, ...controllerProps }}
-                />
-            )}
-        </ReferenceManyFieldController>
+        <ReferenceManyFieldView
+            {...props}
+            {...{
+                children,
```

Comment:
I find it less readable if you explicit all props here, as compared to what we did before.